packages/react/src/react/views/collections/list.client.tsx

```diff
@@ -84,14 +113,63 @@ interface ListTableProps {
 export function ListTable(props: ListTableProps) {
   const navigation = useNavigation()
   const serverFunction = useServerFunction()
+  const queryClient = useQueryClient()
 
-  const [sorting, setSorting] = useState<SortingState>([])
   const [rowSelection, setRowSelection] = useState<RowSelectionState>({})
+  const [searchInput, setSearchInput] = useState('')
+
+  // Debounce the search input by 500ms
+  const debouncedSearch = useDebounce(searchInput, 500)
+
   const [pagination, setPagination] = useQueryStates({
     page: parseAsInteger.withDefault(1),
     pageSize: parseAsInteger.withDefault(10),
+    search: parseAsString.withDefault(''),
+    sortBy: parseAsString.withDefault(''),
+    sortOrder: parseAsString.withDefault(''),
   })
 
+  // Update pagination search when debounced search changes
+  const [prevDebouncedSearch, setPrevDebouncedSearch] = useState(debouncedSearch)
+
+  if (prevDebouncedSearch !== debouncedSearch) {
+    setPagination({
+      page: 1, // Reset page
+      pageSize: pagination.pageSize,
+      search: debouncedSearch,
+      sortBy: pagination.sortBy,
+      sortOrder: pagination.sortOrder,
+    })
+    setPrevDebouncedSearch(debouncedSearch)
+  }
```

Comment:
I expect something like this to group the related logic together

```tsx

useDebounce(searchInput, (value) => {
  // something triggered after debounced
}, 500)
```

With this way, we can remove `prevDebouncedSearch` state from the code